cpp/CMakeLists.txt

```diff
@@ -605,6 +602,7 @@ if(RAFT_COMPILE_LIBRARY)
              ${RAFT_CTK_MATH_DEPENDENCIES} # TODO: Once `raft::resources` is used everywhere, this
                                            # will just be cublas
              $<TARGET_NAME_IF_EXISTS:OpenMP::OpenMP_CXX>
+             $<TARGET_NAME_IF_EXISTS:hnswlib::hnswlib>
```

Comment:
If the user wants to build with HNSW but the target name does not exist will there be an error or will it fail silently? A bad scenario would be to compile the whole library only to find that hnswlib was not included in this compilation when it was explicitly requested.